File: test/test_Data_reader_mark5.py
#!/usr/bin/python

# Copyright (c) 2007 Joint Institute for VLBI in Europe (Netherlands)
# All rights reserved.
#  
# Author(s): Nico Kruithof <Kruithof@JIVE.nl>, 2007
# 
# $Id$

import sys, os,time;

infile = "/jop51_0/kruithof/data/JIVE-13.del"
infile = "data/input.txt"
outfile = "output.txt"

status = os.system("make test_Data_reader_mark5")
if (status != 0): sys.exit(1)

os.system("./test_Data_reader_mark5 "+outfile+"&");
#os.system("Net2file -f "+outfile+"&")

time.sleep(.1)
os.system("File2net -f"+infile)

#print " - Performing diff between files"
time.sleep(.1)
os.system("sync; diff "+infile+" "+outfile)

#print " - Removing output file"
os.remove(outfile)

File: test/test_Data_reader_mark5.cc
/* Copyright (c) 2007 Joint Institute for VLBI in Europe (Netherlands)
 * All rights reserved.
 * 
 * Author(s): Nico Kruithof <Kruithof@JIVE.nl>, 2007
 * 
 * $Id$
 *
 * Tests reading data from a Mark5 (simulation using File2net).
*/

#include <types.h>
#include <Data_reader_mark5.h>
#include <fstream>
#include <assert.h>

#include <stdio.h>
#include <iostream>
#include <unistd.h>

#define BUFFSIZE 1000

int main(int argc, char *argv[]) {
  if (argc != 2) {
    std::cout << "usage: " << argv[0] << " <output_file>" << std::endl;
    exit(1);
  }
  int nBytes;
  char buff[BUFFSIZE];

  Data_reader_mark5 reader("tcp", 2630);
  std::ofstream out(argv[1], std::ios::out | std::ios::binary);

  nBytes= reader.get_bytes(1, buff);
  assert (nBytes == 1);
  out.write(buff, nBytes*sizeof(char));

  nBytes= reader.get_bytes(10, buff);
  assert (nBytes == 10);
  out.write(buff, nBytes*sizeof(char));

  nBytes= reader.get_bytes(100, buff);
  assert (nBytes == 100);
  out.write(buff, nBytes*sizeof(char));

  for (int i=0; i<10; i++) {
    nBytes = reader.get_bytes(50, buff);
    assert(nBytes==50);
    out.write(buff, nBytes*sizeof(char));
  }

  nBytes = reader.get_bytes(BUFFSIZE, buff);
  assert(nBytes==BUFFSIZE);
  out.write(buff, nBytes*sizeof(char));

  int nGet;
  while ((nGet = reader.get_bytes(BUFFSIZE, buff)) > 0) {
    out.write(buff, nGet*sizeof(char));
  }

  out.close();

  return 0;
}
